src/v/iceberg/datatypes_json.cc

```diff
@@ -177,6 +177,9 @@ void rjson_serialize(iceberg::json_writer& w, const iceberg::nested_field& f) {
     w.Bool(bool(f.required));
     w.Key("type");
     rjson_serialize(w, f.type);
+    // TODO: when/if we add support for required fields, 'initial-default' and
+    // 'write-default' should be written here
+    // See  https://iceberg.apache.org/spec/#default-values for requirements
```

Comment:
So this isn't forgotten can we throw if `f.required` is true?

Also can we use the form of `TODO(iceberg)` so they are more greppable.

Reply:
yeah good call

Reply:
done

Reply:
>throw if f.required is true

ah, right the redpanda header struct and sub-fields are still required... my intuition is that we should leave them as such because they are not evaluated with respect to the record schema but I'm not sure.

i'm going to remove the exception from this PR and spin off a separate ticket to avoid blowing up the diff here

Reply:
i might argue that raising the exception from here, in the JSON serde code, is too "low level" for the thing we're trying to flag. policy vs representation or something like that.

Reply:
yeah fair point I wish this was decouples but here we are, I am good with just a comment